pkg/image/apis/image/helper.go

```diff
@@ -494,9 +494,9 @@ func ImageWithMetadata(image *Image) error {
 			image.DockerImageLayers[i].MediaType = layer.MediaType
 		}
 		// reverse order of the layers for v1 (lowest = 0, highest = i)
-		for i, j := 0, len(image.DockerImageLayers)-1; i < j; i, j = i+1, j-1 {
-			image.DockerImageLayers[i], image.DockerImageLayers[j] = image.DockerImageLayers[j], image.DockerImageLayers[i]
-		}
+		// for i, j := 0, len(image.DockerImageLayers)-1; i < j; i, j = i+1, j-1 {
```

Comment:
@legionus when i look at a schema2 image in docker hub (mysql:latest) - oldest layer is first in the manifest layers, top layer is last.  Up until now we didn't need this (because the manifest is upstream pullthrough), I realized as part of this that the ordering was inverted in our version of the manifest.  Can you confirm that schema1 and schema2 *should* do have different orders, or help me identify why this is sufficient to generate a new image correctly?